src/CapabilityService.WebApi/Infrastructure/Messaging/DomainEventRegistry.cs

```diff
@@ -65,7 +65,7 @@ public Type GetInstanceTypeFor(string eventName)
 
             if (registration == null)
             {
-                throw new MessagingException($"Error! Could not determine \"event instance type\" due to no registration was found for type {eventName}!");
+                throw new MessagingHandlerNotAvailable($"Error! Could not determine \"event instance type\" due to no registration was found for type {eventName}!", eventName);

```

Comment:
No tests to prove that we cast MessagingHandlerNotAvailable?

Reply:
And should this implementation be ported to other services using "DomainEventRegistry"?

Reply:
I've added tests in https://github.com/dfds/capability-service/pull/24/commits/2eb021cf3f323e952618ba40dc9e2f1a96455b85 and https://github.com/dfds/capability-service/pull/24/commits/a8a559a1dbe0667eb51ae80e03240872fd43dab2

Reply:
Regarding porting the implementation, I can't see any reason why not, maybe improve the naming of certain Exceptions.

Reply:
I'm open for the idea but it smells sisyphean to me, copying code to and from the various services. Consider whether time can be better spent on proactive infrastructure work instead maybe?